### PR TITLE
FEATURE Option placeholder for upload folder id

### PR DESCRIPTION
**Pull request by tractorcow**
To be merged with https://github.com/silverstripe/silverstripe-asset-admin/pull/1009

### CODE REVIEW

**Comment by kinglozzer** [comment]
Any chance of being able to pass a string, like we can to `->setFolderName()` for UploadFields? Appreciate that may mean some more logic somewhere in this class, but it’d be a nicer dev experience 😄

**Comment by tractorcow** [comment]
I think so; At the JS level it needs to pass the ID, but we could translate the path to id within TinyMCEConfig right?

**Comment by tractorcow** [comment]
@kinglozzer done.

File: src/Forms/HTMLEditor/TinyMCEConfig.php [hunk 1]
```diff
@@ -844,4 +845,19 @@ protected function getAdminModule()
...
     }
+
+
+    /**
+     * Sets the upload folder name used by the insert media dialog
+     *
+     * @param string $folderName
+     * @return $this
+     */
+    public function setFolderName($folderName)
```
Comment (unclecheese) [inline src/Forms/HTMLEditor/TinyMCEConfig.php]: Since it's going into `4`, you can type hint this.

Comment (tractorcow) [reply src/Forms/HTMLEditor/TinyMCEConfig.php]: I thought I did type hint. What form did you want me to use?

Comment (tractorcow) [reply src/Forms/HTMLEditor/TinyMCEConfig.php]: Did you mean `public function setFolderName($folderName) : TinyMCEConfig` ?

**Comment by tractorcow** [comment]
I think I've implemented what you asked @unclecheese :)

**Comment by tractorcow** [comment]
Code is now selfish.

**Comment by unclecheese** [comment]
And can we type the param, too? `string $folderName`?

Sorry to nitpick, but we're just trying to be consistent with all new APIs.

**Comment by unclecheese** [comment]
Nvm, I've done it.

**Comment by tractorcow** [comment]
Thanks @unclecheese . I didn't mean to `string` you out :)

**Pull request merged**